arrows/core/detected_object_set_output_kw18.cxx

```diff
@@ -85,21 +97,64 @@ detected_object_set_output_kw18()
 detected_object_set_output_kw18::
 ~detected_object_set_output_kw18()
 {
+  if( d->m_tot_writer && d->m_tot_writer->is_open() )
+  {
+    d->m_tot_writer->close();
+  }
 }
 
 
 // ------------------------------------------------------------------
 void
 detected_object_set_output_kw18::
-set_configuration(vital::config_block_sptr config)
-{ }
+set_configuration( vital::config_block_sptr config )
+{
+  d->m_write_tot = config->get_value<bool>( "write_tot" , d->m_write_tot );
```

Comment:
Need to add code to merge in defaults. Otherwise get_value will throw if item is not in config block.

```
set_configuration( vital::config_block_sptr config_in )
{
  // Starting with our generated config_block to ensure that assumed values are present
  // An alternative is to check for key presence before performing a get_value() call.
  vital::config_block_sptr config = this->get_configuration();

  config->merge_config( config_in );
```